test/test_dev.py

```diff
@@ -5,13 +5,12 @@
                         unicode_literals)
 
 import os
-import sys
 import re
 from os.path import abspath, dirname, join, relpath
 import shutil
 import pytest
 
-import six
+import six  # noqa F401 'six' imported but unused
```

Comment:
I don't see why six needs to be imported if it's not used. What's the error?

Reply:
Only flake8 check throws an error (F401 'six' imported but unused)
 but the tests pass.

Reply:
We already discussed this case few times. What we did was:
- Remove the import
- If something fails in the test, we check the test failure, we try to understand why this import needs to be here, and we add a the noqa with the error code to skip, and a comment for other devs to know
- If it's difficult for you to understand the error, as often requires a lot of experience, just find the error, post it here, so we can discuss and you can learn about it
- Just adding the error from flake8 doesn't add much value. The first question I'll ask myself if I see this code is "why the import hasn't been removed it it's unused". This is what the code/comment should tell us. Regardless of flake8, why we are importing the module if it's not used

Reply:
sure let me try this